### PR TITLE
docs: Update custom sandbox guide to include steps to use pre-existin…

### PR DESCRIPTION
**Pull request by richardokonicha**
**What problem does this fix or functionality does it introduce?**
Clarifies that users can use pre-existing Docker images from Docker Hub, simplifying the custom sandbox setup process.

**Brief summary of the PR:**
- Updates the custom sandbox guide to include instructions for using pre-existing Docker images.
- Provides an example with the Node.js Docker image.
- Adds steps for creating a custom Docker image and configuring OpenDevin to use it.

**Other references:**
- Fix #2373

Screenshot 2024-07-02 at 2 39 12 PM

### CODE REVIEW

**Comment by richardokonicha** [comment]
@[neubig](https://github.com/neubig) your thoughts and review towards [#2734](https://github.com/OpenDevin/OpenDevin/issues/2734) ?

**Review by Umpire2018 — APPROVED** [review]
You can use `Fix #2373` instead of `Fixes issue https://github.com/OpenDevin/OpenDevin/issues/2734` to correctly link this pr and issue. :)

File: docs/modules/usage/custom_sandbox_guide.md [hunk 1]
```diff
@@ -6,6 +6,25 @@ The default OpenDevin sandbox comes with a [minimal ubuntu configuration](https:
...
 Make sure you are able to run OpenDevin using the [Development.md](https://github.com/OpenDevin/OpenDevin/blob/main/Development.md) first.
 
+
+## Using Pre-existing Docker Images
+
+Creating a custom Docker image is optional. You can also use pre-existing images from Docker Hub, which can simplify and speed up the process. 
+
+### Example with Node.js
+
+If you need a sandbox with Node.js installed, you can specify the desired Docker image directly.
```
Comment (mamoodi) [inline docs/modules/usage/custom_sandbox_guide.md:16]: Where do you specify this?

File: docs/modules/usage/custom_sandbox_guide.md [hunk 2]
```diff
@@ -6,6 +6,25 @@ The default OpenDevin sandbox comes with a [minimal ubuntu configuration](https:
...
+
+sandbox = DockerSSHBox(
+   container_image="node:21"
+)
+````
+
+
 ## Create Your Docker Image
 
 Next you must create your custom docker image, which should be debian/ubuntu based. For example if we want want OpenDevin to have access to the "node" binary, we would use the following Dockerfile:
```
Comment (mamoodi) [inline docs/modules/usage/custom_sandbox_guide.md:30]: Can you please change this to:
"To create a custom docker image, it must be debian/ubuntu based. For example if we want OpenDevin to have access..."

**Review by neubig — COMMENTED** [review]
Hey @richardokonicha , thanks for the contribution! We might want to reword this a bit. Suggestion:

Replace this:

```md
This guide will teach you how to accomplish this by utilizing a custom docker image.
```

with this

```md
There are two ways you can do so:

1. Use an existing image from docker hub. For instance, if you want to have `nodejs` installed, you can do so by using the `node:21` image.
2. Creating your own custom docker image.

If you want to take the first approach, you can skip the first section below.
```

Then we can delete the `Using Pre-existing Docker Images` section that you added anew, since it should be pretty similar.

Finally, in the section where it discusses how to specify the sandbox image, we could again mention that the image might be custom, or it might be a pre-existing image.

**Comment by richardokonicha** [comment]
Screenshot 2024-07-03 at 12 27 04 PM
updated PR #2740 
@neubig @mamoodi review ready.

**Review by neubig — APPROVED** [review]
Looks great, thanks!

**Pull request merged**